tests/e2e/utils/NameGenerator.ts

```diff
@@ -23,4 +29,12 @@ export class NameGenerator {
         return prefix + randomPart;
     }
 
+    public static async getNameFromUrl() : Promise<string> {
```

Comment:
I am not sure that this is the right place for this method

Reply:
Do you have any suggestions for a better place?

Reply:
Maybe a dedicated class?

Reply:
Because obtain name from someone looks like out of scope the "NameGenerator" class

Reply:
What about renaming `NameGenerator` to `WorkspaceNameHandler`? And renaming `generate` method to `generateWorkspaceName`? So the class will have two methods then:

```
class WorkspaceNameHandler {
  generateWorkspaceName() : string;
  getNameFromUrl(): string;
}
```

Reply:
Disagree, not because this is a bad idea, just because better to try don't change a signature and names of the methods and classes, because it can't be applied in the downstream projects automatically.

Reply:
I'm not for creating more classes with just few lines of code when it makes sense to put it together with a similar code. In the downstream you should rely on a specific version of tests for the current version, so you should have enough time to apply changes for the next release. Or am I missing something?

Reply:
I also agree with Katka here. I would prefer to refactor the class into a WorkspaceNameHandler.

Creating a new class just because refactor would break downstream projects doesn't make any sense to me.
Something can introduce breaking changes with every update, that's just what the downstream projects will have to deal with once they update to 7.13.0 .